File: JobFinder-Android/backend/src/main/java/com/deepakmanktala/jobfinder/model/Job.java
package com.deepakmanktala.jobfinder.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

import java.time.LocalDateTime;

@Entity
@Table(name = "jobs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String company;
    private String location;

    @Column(length = 2000)
    private String description;

    private String salary;

    @Column(name = "job_url", length = 1000)
    private String jobUrl;

    private String source;
    private String region;
    private String datePosted;

    @Column(name = "crawled_at")
    private LocalDateTime crawledAt;

    @Column(name = "role_matched")
    private String roleMatched;
}

File: JobFinder-Android/android/app/src/main/java/com/deepakmanktala/jobfinderapp/data/model/Job.kt
package com.deepakmanktala.jobfinderapp.data.model

import com.google.gson.annotations.SerializedName

data class Job(
    val id: Long = 0,
    val title: String = "",
    val company: String? = null,
    val location: String? = null,
    val description: String? = null,
    val salary: String? = null,
    @SerializedName("jobUrl") val jobUrl: String? = null,
    val source: String? = null,
    val region: String? = null,
    @SerializedName("datePosted") val datePosted: String? = null,
    @SerializedName("crawledAt") val crawledAt: String? = null,
    @SerializedName("roleMatched") val roleMatched: String? = null
)

data class JobSearchRequest(
    val roles: List<String>,
    val region: String = "Worldwide (all portals)",
    val dateRange: String = "Last 1 week",
    val concurrency: Int = 3,
    val dedup: Boolean = true
)

data class JobSearchResult(
    val jobs: List<Job>,
    val total: Int,
    val status: String
)

data class PagedJobs(
    val content: List<Job>,
    val totalElements: Int,
    val totalPages: Int,
    val size: Int,
    val number: Int
)
